README.md

```diff
@@ -1,3 +1,182 @@
+
+<a name="setup"></a>
+## [Setup of PHiLIP Environment]("#c_setup")
+The setup will explain how to flash PHiLIP and install the python interface. To setup the PHiLIP environment some hardware will be required, either a nucleo-f103rb or a bluepill with a usb to uart converter.
+
+#### 1. Connect PHiLIP pins
+For the nucleo-f103rb only the usb connection is required. The bluepill will require a uart connection to `A9 - IF_TX` to the uart RX pin and the `A10 - IF_RX` to the uart TX pin. See the [PHiLIP-b pinout]() for more information.  The uart is needed for the basic interface but can also be used for ROM UART flashing.  To flash the bluepill with SWD, connect the swd pins and reset `R - NRST` pin.
+
+#### 2. Flash PHiLIP to the device
+The qualified firmware for PHiLIP is stored in the [PHiLIP repo](https://github.com/riot-appstore/PHiLIP/tree/master/QUALIFICATION/FW).  The correct firmware is needed for the given board, either the [PHiLIP_BLUEPILL](https://github.com/riot-appstore/PHiLIP/tree/master/QUALIFICATION/FW/PHiLIP_BLUEPILL-CURRENT.bin) or the [PHiLIP_NUCLEO-F103RB](https://github.com/riot-appstore/PHiLIP/tree/master/QUALIFICATION/FW/PHiLIP_NUCLEO-F103RB-CURRENT.bin). Previous versions can also be browsed or installed but the _CURRENT_ version is recommended.  To flash the firmware on the nucleo-f103rb, drag and drop the .bin file to the nucleo device.  There are many ways to flash the bluepill, either by connecting a swd connector or with the [ROM UART bootloader](https://medium.com/@paramaggarwal/programming-an-stm32f103-board-using-usb-port-blue-pill-953cec0dbc86).
+
+
+_HINT: If flashing a nucleo-f103rb use the following command_
+
+`wget -P /media/${USER}/NODE_F103RB/ https://github.com/riot-appstore/PHiLIP/raw/master/QUALIFICATION/FW/PHiLIP-NUCLEO-F103RB-CURRENT.bin
+`
+
+#### 3. Install the Python Interface (philip_pal)
+The philip_pal is only python3 so use the following command to install:
+
+`sudo pip3 install philip_pal`
+
+<a name="gs_dev"></a>
+## [Getting Started as a Developer]("#c_gs_dev")
+_First follow the [setup]("#setup")_
+
+To used PHiLIP as a developer, an interface shell `philip_shell` is provided with the [philip_pal package](IF/philip_pal/README.md).  The philip_shell has an connect wizard, command history, and auto-completion.  If in doubt, try pressing tab a few times.
```

Comment:
s/used/use

Reply:
heh, listening to music with words and writing without proofing is maybe not the best combo 😆